java/client/src/main/java/glide/api/commands/GenericBaseCommands.java

```diff
@@ -590,4 +591,76 @@ CompletableFuture<Boolean> pexpireAt(
      * }</pre>
      */
     CompletableFuture<Boolean> copy(String source, String destination, boolean replace);
+
+    /**
+     * Serialize the value stored at <code>key</code> in a Redis-specific format and return it to the
+     * user.
+     *
+     * @see <a href="https://valkey.io/commands/dump/">valkey.io</a> for details.
+     * @param key The key of the set.
+     * @return The serialized value of a set.<br>
+     *     If <code>key</code> does not exist, <code>null</code> will be returned.
+     * @example
+     *     <pre>{@code
+     * byte[] value1 = client.dump("myKey").get();
+     * assert value1.equals("value1");
+     *
+     * byte[] value2 = client.dump("nonExistingKey").get();
+     * assert value2.equals(null);
+     * }</pre>
+     */
+    CompletableFuture<byte[]> dump(byte[] key);
+
+    /**
+     * Create a <code>key</code> associated with a <code>value</code> that is obtained by
+     * deserializing the provided serialized <code>value</code> (obtained via DUMP).
+     *
+     * @see <a href="httpshttps://valkey.io/commands/restore/">valkey.io</a> for details.
+     * @param key The key of the set.
+     * @param ttl The expiry time (in milliseconds). If 0, it means creation without any expiry.
+     * @param value The serialized value.
+     * @return Return <code>OK</code> if successfully create a <code>key</code> with a <code>value
+     *      </code>. Return a "Target key name is busy" error when <code>key</code> already exists
+     *     unless use the <code>REPLACE</code> modifier. If RDB version and data checksum don't match,
+     *     an error is returned.
+     * @example
+     *     <pre>{@code
+     * String value1 = client.restore("newKey", 0, "value").get();
+     * assert value1.equals("OK");
+     * }</pre>
+     */
+    CompletableFuture<String> restore(byte[] key, long ttl, byte[] value);
+
+    /**
+     * Create a <code>key</code> associated with a <code>value</code> that is obtained by
+     * deserializing the provided serialized <code>value</code> (obtained via DUMP).
+     *
+     * @see <a href="httpshttps://valkey.io/commands/restore/">valkey.io</a> for details.
+     * @param key The key of the set.
+     * @param ttl The expiry time (in milliseconds). If 0, it means creation without any expiry.
+     * @param value The serialized value.
+     * @param restoreOptions The {@link RestoreOptions}.
+     * @return Return <code>OK</code> if successfully create a <code>key</code> with a <code>value
+     *      </code>. Return a "Target key name is busy" error when <code>key</code> already exists
+     *     unless use the <code>REPLACE</code> modifier. If RDB version and data checksum don't match,
+     *     an error is returned.
+     * @example
+     *     <pre>{@code
+     * String value1 = client
```

Comment:
You don't have to follow spotless in the example - you can remove [some] line breaks.
I'd also define a variable for options:
```
RestoreOptions options = ...
String response = client.restore(...)
assert ...
```